src/cockpit/389-console/src/lib/database/databaseConfig.jsx

```diff
@@ -990,3 +990,580 @@ GlobalDatabaseConfig.defaultProps = {
+            log_cmd("handleSaveDBConfig", "Applying config change", cmd);
+            cockpit
+                    .spawn(cmd, { superuser: true, err: "message" })
+                    .done(content => {
+                        // Continue with the next mod
+                        this.save_ndn_cache(requireRestart);
+                    })
+                    .fail(err => {
+                        const errMsg = JSON.parse(err);
+                        this.props.reload(this.state.activeTabKey);
+                        this.setState({
+                            saving: false
+                        });
+                        this.props.addNotification(
+                            "error",
+                            cockpit.format(_("Error updating configuration - $0"), errMsg.desc)
+                        );
+                    });
+        } else {
+            this.setState({
+                saving: true
+            }, () => { this.save_ndn_cache(requireRestart) });
+        }
+    }
+
+    render() {
+        let spinner = "";
+        if (this.state.loading) {
+            spinner = (
+                <div className="ds-loading-spinner ds-margin-top-xlg ds-center">
+                    <TextContent>
+                        <Text component={TextVariants.h3}>
+                            Loading global database configuration ...
+                        </Text>
+                    </TextContent>
+                    <Spinner className="ds-margin-top" loading size="md" />
+                </div>
+            );
+        }
+
+        let saveBtnName = _("Save Config");
+        const extraPrimaryProps = {};
+        if (this.props.refreshing) {
+            saveBtnName = _("Saving config ...");
+            extraPrimaryProps.spinnerAriaValueText = _("Saving");
+        }
+
+        return (
+            <div className={this.state.saving ? "ds-disabled ds-margin-bottom-md" : "ds-margin-bottom-md"} id="db-global-page">
+                {spinner}
+                <div className={this.state.loading ? 'ds-fadeout' : 'ds-fadein'}>
+                    <TextContent>
+                        <Text className="ds-config-header" component={TextVariants.h2}>
+                            {_("Global Database Configuration")}
+                        </Text>
+                    </TextContent>
+
+                    <div className="ds-margin-top-lg">
+                        <Tabs isFilled activeKey={this.state.activeTabKey} onSelect={this.handleNavSelect}>
+                            <Tab eventKey={0} title={<TabTitleText>{_("Database Size")}</TabTitleText>}>
+                                <div className="ds-left-indent-md">
+                                    <Grid
+                                        title={_("Database maximum size in bytes. The practical maximum size of an LMDB database is limited by the system’s addressable memory (nsslapd-mdb-max-size).")}
+                                        className="ds-margin-top"
```

Comment:
I noticed several fields are very close to bottom of the tabs, I think it would look a little nicer if you used this class instead:  ds-margin-top-xlg

Do this for all the grids below so they are consistent, thanks!

Reply:
I agree, it looks way better